### PR TITLE
Fix correlation ID propagation in SQS v2 publisher

### LINKED ISSUE #453: Correlation ID is not propagated with SQS v2 publisher
## Is your feature request related to a problem? Please describe

The SQS v2 publisher does not add the `correlationID` to messages, which is a different behaviour from the [previous version](https://github.com/beatlabs/patron/blob/master/client/sqs/publisher.go#L52-L77) of the publisher.

## Describe the solution

Add `correlationID` as a message attribute to the v2 publisher.

### PR DESCRIPTION
**Pull request by ashraymehta**
## Which problem is this PR solving?
Resolves #453 (Correlation ID is not propagated with SQS v2 publisher).

## Short description of the changes

Adds `correlationID` as a message attribute to the SQS v2 publisher. Avoids adding this header if it's already set.

### CODE REVIEW

**Review by mantzas — DISMISSED** [review]
LGTM

**Comment by ashraymehta** [comment]
@mantzas Could you please approve running of GitHub workflows as well? CI checks are waiting. :)

File: client/sqs/v2/publisher.go [hunk 1]
```diff
@@ -83,7 +84,7 @@ func (c sqsHeadersCarrier) Set(key, val string) {
 }
 
 // injectHeaders injects the SQS headers carrier's headers into the message's attributes.
```
Comment (gmaz42) [inline client/sqs/v2/publisher.go]: Shall we update GoDoc to mention that it will inject the correlation ID only if not already present?

Comment (ashraymehta) [reply client/sqs/v2/publisher.go]: I've modified the godoc. Could you take another look? I'm unable to re-request a review from you on GitHub for some reason. :/

**Review by mantzas — APPROVED** [review]

**Review by gmaz42 — APPROVED** [review]

**Review by tpaschalis — APPROVED** [review]

**Comment by ashraymehta** [comment]
Thanks for the approvals! Could someone with write access please merge the PR? 🙏

**Pull request merged**